### DISCUSSION TITLE
Possibility to disable comment suggestion?

### DISCUSSION THREAD

**Discussion by lehtiniemi** [community/community#8062] [Copilot Conversations]
This product is really incredible. The one thing I would like to have is the ability to disable these two behaviors upon will:
-disable comment auto-complete - I like to write my comments myself and a suggestion that appears on screen distracts the thought process because I automatically read it
-disable suggestion of functions when nothing is being typed. I find that when I'm inside a class between methods, CoPilot starts suggestion full methods without nothing being typed. I don't want this because I constantly have to hit esc to stop suggestions. Please see the attachment. This sometimes happen, sometimes not (don't know the reason yet).

Both are related to CoPilot trying too much and this requires my constant attention and becomes a distraction.

But I really like the method and code line autocomplete, they save a ton of time so just some checkboxes to limit the feature functionality would be nice.

**Comment by jaquinocode** [comment]
Just chiming in to say that the only gripe I have with Copilot is that inline suggestions show while I'm writing a comment. It distracts my thought process and slows me down significantly.

Without that issue, Copilot is game-breaking. But with that issue, it's kind of a double-edged sword in terms of improving your productivity.

**Comment by jaquinocode** [comment]
Here's a workaround I figured out for how to fix the comment suggestions problem! 🚀 

The basic idea is to create a keyboard shortcut that we press in order to disable the inline suggestions from popping up. So before we write a comment, we can just press the keyboard shortcut to disable suggestions, then write the comment, and then press the keyboard shortcut to enable suggestions again. It's not a perfect solution, but it works!

To do this:
1. Install [the Toggle extension](https://marketplace.visualstudio.com/items?itemName=rebornix.toggle&ssr=false#overview).
    - (It's written by a vscode contributor and enables you to toggle settings with keyboard shortcuts. We need to do this because the Copilot extension doesn't actually provide a keyboard shortcut to disable inline suggestions.)
2. To your keybindings file for vscode, add the keyboard shortcut snippet below. You only need to change the values for `key` in order to have it work. (As of the time this comment was written, `github.copilot.inlineSuggest.enable` is the setting we need to toggle here. If they ever change that setting name in the future you'd have to change the below snippet accordingly.)
```jsonc
{
    "key": "shift+cmd+c",  // set this to whatever works for you
    "command": "toggle",
    // By the way, this "when" expression was inspired from the "when" expression for Copilot
    // extension's "Trigger Inline Suggestions" keyboard shortcut.
    "when": "editorTextFocus && !editorHasSelection",
    "args": {
      // This id is just a unique name you come up with yourself
      "id": "toggleGithubCopilotInlineSuggestionsBeingEnabled",
      // The names and values of the setting you want to toggle. In this case, it's to enable/disable the
      // `github.copilot.inlineSuggest.enable` setting.
      "value": [
        {
          "github.copilot.inlineSuggest.enable": true
        },
        {
          "github.copilot.inlineSuggest.enable": false
        }
      ]
    }
},
```
For more info on how to setup a toggle keyboard shortcut you can look at the extension page for the Toggle extension. You can also read [more info about it here](https://medium.com/hack-visual-studio-code/toggle-any-setting-in-vs-code-using-keyboard-shortcut-arguments-cdb5ddc56955).

3. Press the keyboard shortcut you set up and notice that the corresponding setting in vscode's settings file changes in response. Cool, you're done!

Note: This will allow you to disable inline suggestions from showing up. But it won't make a current inline suggestion, that's already showing, vanish. Just press Esc in that situation if it ever comes up. Ideally, you'd disable inline suggestions ahead of time before they'll pop up. That way you'll never have to press Esc.

**Reply by darajava** [reply]
This is not a solution

**Reply by alex-calderwood** [reply]
I do something similar to this but I did it without a template since I never mess with the settings.json. You can just bind a hotkey to toggling the completions using the UI.

I think this is the best short-term fix, but I wish it would automatically detect whether we are in a comment (for whatever language) and not serve a completion. 

However, there are some problems with this request - there are a few times that I want it to write documentation for me... so this temporary solution of manually turning off/on completions actually handles that pretty well.

**Reply by b-hayes** [reply]
Rather just never get suggestions inside comments then to have to keep turning it on and off.

**Reply by DorianNaaji** [reply]
This is suboptimal but unless we don't have a real implementation change from GitHub and their plugin, this is the only way not to be distracted by comment suggestions.

_I ended mapping my IntelliJ IDEA Github Copilot "toggle completion" to a convenient shortcut (`Shift+G`).
And I just toggle it off when writing a comment, and then back on..._

_Setting is : `Copilot: Disable Completions` which defaults at `Ctrl+Alt+Shift+O`._

**Comment by jaquinocode** [comment]
Github Copilot extension contributors, if you were to provide a keyboard shortcut to toggle enabling inline suggestions then the above instructions wouldn't be needed. I highly suggest you provide one!

**Comment by jamesonknutson** [comment]
@jaquinocode @lehtiniemi 

I too got tired of copilot throwing suggestions in my face and breaking up my mental flow, [so I wrote a VSCode extension that makes Copilot STFU when you're writing comments.](https://marketplace.visualstudio.com/items?itemName=disable-copilot-comment-completions.disable-copilot-comment-completions) It might be a better solution than your hotkey fix @jaquinocode

**Reply by jaquinocode** [reply]
Neat, I'll take a look!

**Reply by steinhh** [reply]
But is it possible to configure this to prevent comment suggestions at all? Copilot often suggests "//" after the end of a statement, i.e. "would you like to write some comment about what this line does". But that's not how you should write code ;-). 

I've tried adding "\\;" to the inhibitMatchers setting, but it seems not to work, am I just doing something wrong?

**Reply by sergiotasconmorales** [reply]
Very useful. I was tired of copilot trying to (unsuccessfully) read my mind every time I tried to write a comment :'D Thanks @jamesonknutson

**Reply by ryanberckmans** [reply]
great extension, thanks.

**Reply by jarrodwhitley** [reply]
Anyone out there capable and willing to make a similar plugin for PhpStorm?

**Reply by ErikRossRonnow** [reply]
you're goated bro thanks, just what i was looking for aswell, so cringe when it autocompletes a comment and you completely lose your train of thought! <3

**Comment by verhovsky** [comment]
Instead of just disabling comments suggestions it could be better to increase the threshold of confidence required to show the suggestion when you detect that the suggestion is entirely commented out text, because when Copilot completes a long variable names in a comment is useful.

**Comment by JoannaFalkowska** [comment]
I would love to see this feature. In 99% cases the comment suggestions are incredibly disruptive, useless and annoying. In the remaining 1% they're still disruptive and useless, but at least they're so absurd they become funny.

![image](https://user-images.githubusercontent.com/14551750/170727775-5affe001-187e-4a81-a334-72c965c9a0a6.png)
![image](https://user-images.githubusercontent.com/14551750/170727849-8cf9b631-083a-4cad-b0c7-5c88d64b8d78.png)

![image](https://user-images.githubusercontent.com/14551750/170727944-977a7ba7-2658-4c89-a21c-66b1ab8d5291.png)

**Comment by sudara** [comment]
This is my #1 feature request. We should be able to disable comments.

When writing a comment, I have the "intent to clarify." There's 0 cases out of hundreds where Copilot correctly guessed this intent (IMO this isn't possible by definition), so it's either a  source of amusement or noise, and devalues what it's actually good at.

**Reply by iandunn** [reply]
I have exactly the same problem. I think I'm gonna disable CoPilot because these are so distracting and happen so often. It far outweighs the number of times it suggests a useful line of code.

**Reply by steinhh** [reply]
And as someone pointed out somewhere else: **If** your code is so clearly written that an artificial intelligence is able to write your comment,, a human intelligence will be able to understand your code as well: ergo, the comment is not needed and should not exist!

**Comment by StephenHaney** [comment]
Big +1... the comment suggestions are hilariously wrong and really distracting. I haven't had a single one that's even related to what I want to actually write. And my brain auto-reads whatever pops up, which hijacks my thought process. I'd love the ability to disable inline suggestions for all comment blocks. I'm disabling copilot for now until this is supported.

**Reply by jamesonknutson** [reply]
I'm not sure if you're using Jetbrains or some other similarly-not-vscode editor, but I just updated my extension I wrote (referenced above) to allow user configuration of the scopes where Copilot's suggestions should be disabled, in case that fixes your issue.

**Reply by haszari** [reply]
It's like a teenager, constantly interrupting your thoughts!

**Comment by floer32** [comment]
I'd really like to disable comments-completion in JetBrains editors as well.

**Comment by tbenthompson** [comment]
+1 on this. Very distracting. Is providing an option here on a roadmap anywhere? I like Copilot enough to keep it enabled but I find myself looking away from the computer screen while I type longer comments!! I guess it's good typing accuracy practice.

**Comment by aaroncrawford** [comment]
This is single-handedly keeping me from buying copilot.  I comment a lot and this breaks my flow entirely.

**Reply by sergiotasconmorales** [reply]
For me the extension by @jamesonknutson has been useful as a temporary solution. Now I don't get any suggestions for comments

**Comment by ryanjsalva** [comment]
Thanks for the feedback, y'all. What I'm hearing is:

* **Problem:** Copilot is too "chatty" when the developer is writing code comments
* **Workarounds:** To cut-down on the noise, a few folks have used other extensions to toggle Copilot on/off via keyboard shortcuts 
* **Ask:** Build some gesture into the Copilot extension that reduces noise when writing code comments

**Reply by iandunn** [reply]
Rather than reducing noise, I'd prefer an option to disable Copilot comments entirely

**Reply by tbenthompson** [reply]
Yes, I’ve almost never had an example of a comment where Copilot was useful. Turning Copilot off entirely for comments would be by preference.

**Reply by steinhh** [reply]
Turning it entirely off is the best for me too -  rephrasing myself: If Copilot can figure out the comment from the code, the comment is unnecessary. If a comment is necessary, Copilot can't possibly figure it out (yet) 😆.

But for those who don't want to turn it off, my guess is that they would like a decrease in chattiness.

Quite unrelated: 

To spare my left hand/arm from twisting up to the tab key too often, I've made *enter = accept*. Of course, this exacerbated the EOL comment issue, having to press escape (or do other gymnastics) to avoid accepting the EOL comment.

Now I'm experimenting with *enter = accept* **and** *alt-enter* (and *cmd-enter*) = force new line (minimal twist with right hand). It can be vice versa, of course, but I accept suggestions more often than I start a new line (even more so when EOL comments are a thing of the past!). Anyway, my point here is that *recording keyboard shortcuts that include enter is impossible in VSCode* - because enter is how you end a shortcut recording! But you can do it in the keybindings.json file! This took me a huge amount of digging to find out (giving up on at least one attempt), so to spare others: maybe there's a good place for this information? Not just a pointer to "how to record keystrokes", 'cause that doesn't cut it😆. Yup, this is really a VSCode thing, but I think it would be very beneficial to have this tip closer at hand...

**Reply by aaroncrawford** [reply]
I'd also prefer disabling it for comments entirely.

**Reply by sudara** [reply]
+1 to complete disabling, I'd rewrite the ask like so:

**Ask:** Add a feature toggle to the VSCode, Jetbrains and other Copilot integrations to never autocomplete when in a comments context.

**Reply by xvdp** [reply]
I agree with everyone about thought process disruption. I would also like to be able to toggle copilot altogether sometimes. 
* Comments off most often please. The only time when I find them useful is when I have a whole bunch of classes already done so it may copy my syntax for Args perhaps.

* Toggle copilot on and off completely
- For example Im writing a skeleton class, or Im writing something I know very well how to and I like my coding style- I prefer to just go.

Maybe this could be implemented like pylint mutes e.g. (i dont like digging settings, I dont like key shortcuts)
#pylint: disable=no-member 
something like? 
#copilot: disable=comments
#copilot: disable=this-class

**Reply by xvdp** [reply]
Addendum to my comment. Copilot keeps trying to get me to do code that I don't want. 
It is super useful in the case when I'm trying to do something that im not completely familiar - for instance, if I'm writing some code to do with camera calibration and I don't remember the distortion parameters; or in the typical standard libraries where which one doesn't exactly remember order of args or name of kwargs 

* Disable extension ( I'm in 'nightly' and the latest vscode) DOES NOT WORK - it keeps suggesting stuff.
I would program mostly on disable mode, then enable it when I'm groping.

* or extending my pylint syntax analogy: perhaps a simpler switch syntax. eg. Disabling globally and locally enabling - inside a function, class or file
#copilot: enable #to enable in this project, file, class, function, perhaps overriding a prior 
#copilot: disable #same as above

**Reply by d2x** [reply]
I too would like the ability to disable comment auto-completion in JetBrains IDEs completely. It's great if it wants to suggest a block of code based on comments, but the comments are meant to be for my own use when mapping out workflows that need to be achieved. Even prefixing them as `// TODO:` doesn't stop it from suggesting things which then breaks concentration.

+1 for **Ask**: Add a feature toggle to the VSCode, Jetbrains and other Copilot integrations to never autocomplete when in a comments context.

**Reply by AndrewBelous** [reply]
Would love to see "no completions in comments" be a setting for Copilot. Just upvoting with my comment. Please make it happen? So incredibly distracting to see completions in my comments.

**Reply by iandunn** [reply]
@ryanjsalva Can you give us an update on this? There's a large demand for this because CoPilot's comments are very disruptive and frustrating. It seems like it'd be very simple to add a setting to disable comments.

Have y'all made any decisions about this? Is there a timeline to implement something?

**Comment by ewhitesides** [comment]
perhaps copilot could be set so if you start the line double comment chars (for example ## instead of just #), copilot would know not to suggest anything? that would allow for in-the-moment disabling without having to start/stop the extension.

**Comment by joliss** [comment]
I do occasionally have situations where Copilot produces useful completions in comments, so I don't want to completely disable it in comments. But I agree with the original poster that it's distractingly noisy at the moment.

So I think that ideally, Copilot would suggest completions in comments only when it's reasonably sure about them.

**Comment by jesseconnr** [comment]
This really needs to be an option. I just started trialing Copilot and I've already had to disable inline autocompletion on everything solely because of the distraction it causes when autocompleting comments. The existing workarounds aren't great for a paid product.

**Comment by TravisBumgarner** [comment]
Adding a +1 to this. I find myself turning off copilot when it disrupts my thinking while writing comments. Then I forget it's off for hours or days.

**Comment by WesleyKapow** [comment]
<img width="261" alt="Screenshot 2023-06-09 at 11 49 54 AM" src="https://github.com/community/community/assets/1568246/5a0f8d9e-d2ff-4a5f-9a30-b8ad17a61596">

Thanks Copilot 😛 

Seriously though, the comments have a much lower hit rate than code suggestions and I find them very distracting. I tried some of the solutions mentioned here but nothing worked. Please allow disabling of comment suggestions!!!

**Comment by ablegrape** [comment]
This is my #1 problem with Github copilot. The comment suggestions drive me absolutely nuts. I would like to be able to disable them in IntelliJ IDEA.

Github, are you listening? Is anything in progress?

**Comment by xvdp** [comment]
I stopped using copilot on that account. While auto coding looks like the
future in many respects, without interaction and instructions as simple as
-- dont @#$!@% comment my code -- it doesnt look like copilot wants part of
it, or?

**Comment by Fishrock123** [comment]
Copilot sometimes suggests useful code, but often suggests very circular comments when I am trying to explain code (which usually only happens when code is non-trivial).

This functionality should be able to be enabled/disabled separately.

**Comment by johnrantala** [comment]
<img width="372" alt="image" src="https://github.com/community/community/assets/141130335/eacb9185-fd38-40b1-8a0d-ae34e654003f">

Using Visual Studio Pro 2022, I concur with the feedback on disabling code comments— they're distracting with no current workaround in Visual Studio.

Occasionally, I'd like to turn off Copilot with a hotkey anyways, especially when defining variables since it often misinterprets my intent. However, once variables are declared, Copilot performs well. There's no quick hotkey to toggle Copilot without a mouse in Visual Studio afaik.

**Comment by dodobyte** [comment]
does anyone care? issue is here for over a year and this should be very easy to fix. comment suggestions are horrible and distracting.

**Reply by ryanberckmans** [reply]
I use this extension to prevent copilot from suggesting comments:

https://marketplace.visualstudio.com/items?itemName=disable-copilot-comment-completions.disable-copilot-comment-completions

If you like the extension, please leave a 5-star review (I am not the author but I want it to be maintained).

Out of the box by default, this extension prevents copilot from suggesting inline comments in source files.

I also use these settings to prevent copilot suggestions in commit msgs (as I use vsc as my EDITOR for git commit)

```json
  "disable-copilot-comment-completions.globPatterns": [
    "**/COMMIT_EDITMSG"
  ],
```

**Comment by haszari** [comment]
The comment suggestions are like a teenager, constantly interrupting with inane, unhelpful stuff they found on the internet. An option to disable comment completion suggestions (or put them behind a dropdown/key etc) would be a huge improvement.

**Comment by tim816** [comment]
I'm quite surprised this hasn't been addressed already.   Please consider adding a toggle to disable comment completions entirely. The way that I provide context to copilot is by adding a comment describing the intent. The robot should not be suggesting to me the intent. It does produce very helpful suggestions for the code itself. On the other hand, the comments are just ridiculous.

At best, comment completion is distracting, but most of the time it's downright annoying and inaccurate. I have not seen it produce a helpful comment that I hadn't basically finished typing. 

In society, do we not consider it rude to _constantly_ finish another's sentence in a conversation?

**Comment by fahmizainal17** [comment]
Could you guys help me get two answer accepted?

**Comment by ToM-Korn** [comment]
+1

**Comment by enistola** [comment]
+1

**Comment by aaroncrawford** [comment]
Can we get an explanation why this is either A) difficult to implement or B) not seen as an important option?

Copilot has been great except the comment suggestions.  My team comments a lot and this breaks flow entirely.  The success rate is probably 1%.  We'd much rather be in charge of comments, then have Copilot attempt to create the line/block based on our comments rather than it just wildly guessing at what we're attempting to do.

**Comment by ToM-Korn** [comment]
I found a quick workaround for myself... Maybe it helps someone else as well. 

1. I just put (in python) another # at the end of the line. 
2. Put the Cursor in front of it 
3. Then write my comment between the two ##

`# I write a comment#`

Copilot does not write comments if you are in the middle of the text 
![copilot](https://github.com/community/community/assets/1007437/6a8bfed2-f772-45c3-9cc8-d60396c98a7e)

**Reply by koynoyno** [reply]
Awesome advice. Thank you.

**Comment by iandunn** [comment]
If GH doesn't want to allow completely disable the processing of comments, then as a compromise they could add a option so that they're only processed with a prefix.


This comment _would_ be processed:
```js
// copilot loop through the response and create a DOM element for each item, then append it to the container
```


This comment _wouldn't_ be processed:
```js
// todo refactor this after launch
```

That'd work similar to how `ok google` or alexa only respond when explicitly prompted.

**Comment by steinhh** [comment]
The disable-copilot-comment-completion plugin works quite well!

The only confusing thing is that it changes the .vscode/settings.json file to set "github.copilot.inlineSuggest.enable": false whenever you're in a comment. Then it changes back when you're in a piece of code. (I guess it's VSCode itself that changes the file, as a result of the setting being toggled)

**Reply by ablegrape** [reply]
Also good to remember that VSCode isn't the only platform for Copilot. (There's IntelliJ IDEA, for example.) Unless there are plugins for every platform, it would be much better for the comment completion disabling to be part of Copilot itself.

**Comment by AuthorOfTheSurf** [comment]
2024 —

There is a certain Zen to finding the right moment to comment, and writing a quality comment that truly serves its purpose

The constant auto-suggested comment text is driving me crazy because CoPilot is wrong regarding comments 99% of the time

Business context, nuance, and precise control of ordering and level of detail is required for comments to have their intended impact

I use WebStorm (same as IntelliJ) by JetBrains, so VSCode-specific solutions do not work for me

**Comment by ethel-dev** [comment]
This is wrong how long it has taken to address something as silly as this... VSCode user for years... Let humans turn off the AI's annotation of the code. We use AI to write tedious code that people have done millions of times, not to understand and intuit purposes that the AI almost always knows very little about. The AI cannot understand the assignment right now like a human can.

**Comment by gregb-jobber** [comment]
+1. The comment suggestions completely break my thought process and they're not as helpful as the code suggestions.

Also some times it start the comment with `// This is a hack ...` and I feel attacked. 😆

**Comment by chreekat** [comment]
I use https://github.com/github/copilot.vim. The inability to disable comment suggestions made Copilot pretty frustrating. Luckily, I was just able to figure this out. I haven't messed with it much but it seems to help:

```patch
diff --git a/autoload/copilot.vim b/autoload/copilot.vim
index 0236325..c623233 100644
--- a/autoload/copilot.vim
+++ b/autoload/copilot.vim
@@ -408,6 +408,15 @@ function! copilot#IsMapped() abort
 endfunction

 function! copilot#Schedule(...) abort
+  " Only fire if not in a comment. Have to step back one column -- see note at
+  " :help synID().
+  " Caveat 1: Copilot will still fire if the cursor is on a blank line. -_-
+  " Caveat 2: "[synID] can be very slow."
+  let synName = synIDattr(synIDtrans(synID(line("."),max([1,col(".")-1]),1)),"name")
+  if synName ==# "Comment"
+    call copilot#Clear()
+    return
+  endif
   if !s:has_ghost_text || !copilot#Enabled() || !copilot#IsMapped()
     call copilot#Clear()
     return
```

**Reply by svermeulen** [reply]
This didn't work for me.  I also tried a few more variations of this to no avail, so just used this terrible hack instead:

```diff
diff --git a/autoload/copilot.vim b/autoload/copilot.vim
index 782183f..0a91580 100644
--- a/autoload/copilot.vim
+++ b/autoload/copilot.vim
@@ -405,8 +405,24 @@ function! s:Trigger(bufnr, timer) abort
   return copilot#Suggest()
 endfunction
 
+lua << EOF
+function copilot_vim_is_on_comment()
+    local current_line = vim.fn.getline('.')
+    local filetype = vim.bo.filetype
+
+    if filetype == "lua" then
+        return current_line:match("^%s*%-%-") ~= nil
+    elseif filetype == "cs" then
+        return current_line:match("^%s*//") ~= nil
+    end
+    -- TODO add more filetypes
+
+    return false
+end
+EOF
+
 function! copilot#Schedule() abort
-  if !s:has_ghost_text || !s:Running() || !copilot#Enabled()
+  if !s:has_ghost_text || !s:Running() || !copilot#Enabled() || v:lua.copilot_vim_is_on_comment()
     call copilot#Clear()
     return
   endif
```

**Reply by steinhh** [reply]
How/where do you apply that?

**Comment by davjs** [comment]
@ryanjsalva It's 2024 and this issue remains unresolved. 

I can only assume the delay for fixing this is due to expectations that Copilot will over time provide more helpful comments.

However, it's important to note that the concern isn't solely about verbosity but rather about accuracy in understanding user intent. 
When users provide comments, they often do so to convey context beyond what can be inferred directly from the surrounding code.
This context is not yet accesible to the AI and it will likely be a long time until it can be inferred. Until this time comes (if it ever does) I propose you get of your high horses and make a practical fix for this issue. That way us who like reap the benefits of github copilot while maintining high quality comments can do so without disruption.

// Unsubscribing yet again

**Comment by duc2019** [comment]
I disabled copilot due to this issue despite my company purchased the enterprise license.
I assume copilot is only for noobs that they can't even write their own word down.
Good engineers able to write comments themself is not likely to use copilot, I think this is the reason they keep this issue open for 3 years.

**Comment by ablegrape** [comment]
Kinda nuts that 3 years on, this issue, which is *by far* my number 1 gripe with Copilot, and sounds like it is for many others, is still not only unresolved, but hasn't had any comments or acknowledgment from the Github team in more than a year and a half (and thank you, @ryanjsalva, for that, when you did). It would be nice to have some kind of idea of whether this is even on the radar, and if so, for what kind of schedule.

**Comment by peabnuts123** [comment]
For those struggling with this, I personally have found the keybind `GitHub Copilot: Enable/Disable Copilot completions (github.copilot.toggleCopilot)` a solid workaround for this. I toggle copilot on/off all day (not just due to comments). I think that's a sensible use of a tool. You don't want to use it for everything all the time. I would still love a setting to outright disable it suggesting while you are in a comment block, but using the shortcut to turn it on/off is an acceptable workaround IMO.

**Comment by FacioRatio** [comment]
Just throwing my voice into the wind. The day an AI can write a comment for me is the day I'm obsolete, so I'll never need this distracting feature. Attention is the only currency.

**Comment by cwoodhayes** [comment]
+1 pls @ryanjsalva

**Comment by weswaugh** [comment]


**Reply by steinhh** [reply]
It says it as it should have been... thanks for the report!

**Reply by peabnuts123** [reply]
There should be a cog in the bottom left of your screen … ? That's where settings is 🤔

**Reply by peabnuts123** [reply]
It looks like maybe the size of your editor or an extension / setting / something has collapsed your sidebar to a horizontal list of tabs above the word "Explorer". The cog in your screenshot is in the top-right, next to minimize et al. 

My vscode looks like this, by default:
<img width="971" alt="image" src="https://github.com/user-attachments/assets/139705fa-66a4-4c36-a622-aa797d0838be">

**Comment by mbylstra** [comment]
If AI is so good at writing code like Github's report says it is, shouldn't this be a trivial feature to implement by the Github team? Clearly no one wants autocomplete in comments. How is it still here?

**Comment by daturkel** [comment]
I had disabled all inline suggestions but was still seeing the prompt to press command-I to continue with copilot. I was finally able to disable it using the experimental setting: Features > Chat > Inline Chat: Line Natural Language Hint (akak inlineChat.lineNaturalLanguageHint). This is in Code Insiders.

**Reply by mvasilkov** [reply]
This is fantastic, thanks!

My settings.json file is 90% dedicated to turning off endless garbage lol.

**Reply by TroldeJens** [reply]
This works. Excellent!

**Comment by enreda-net** [comment]
I found the best solution for me.
Uninstall GitHub Copilot, they rarely help me generate code, I have discovered that unfortunately I have been wasting time with this garbage installed.

In my case I have managed to take advantage of almost 1% of the time writing code without help, although it is true that the blame for this is the time lost by annoying suggestions in the comments.

Goodbye GitHub Copilot

**Reply by b-hayes** [reply]
I just use the shortcut key for turning it off and on. It remains off 90% of the time now and I just turn it on when I know Im about to types some predictable and repetitive code.

**Comment by DylanWhitaker** [comment]
I've found a workaround is to type an uncensored swear word at the start of the comment, e.g. "Copilot **** off." FWIW I think it's the swear word itself that's stopping the suggestions, but personally I find my way more satisfying.

**Reply by steinhh** [reply]
WTF😂? Does that actually work? Does "WTF" work too😆?

**Reply by DylanWhitaker** [reply]
Anecdotal, but it's worked for me so far. I assume there's a profanity filter.
![image](https://github.com/user-attachments/assets/dc37f493-5d45-4ddb-b1b4-f0abdfcb35b0)

As for "WTF", apparently not 😄 
![image](https://github.com/user-attachments/assets/408481fe-82ec-428b-9444-7c21e8130e57)

**Reply by marktheunissen** [reply]
GENIUS!!

**Reply by apilatosba** [reply]
this works haha 🤠🤠

**Comment by mattparlane** [comment]
Just adding my voice to the chorus of developers asking for this. I don't see why this is taking so long.

https://www.youtube.com/watch?v=Vhh_GeBPOhs

**Comment by Rippletank** [comment]
The issue I have is with overly obvious comments that are part of an otherwise correct suggestion for the next line. Like these I'm looking at right now:
wgpu::Buffer vertexBuffer; // Buffer for GUI vertices
const uint32_t styleBinding = 1; // Binding index for styles
It's more readable without them and the speed up of autocomplete is negated by having to delete the comment. 
Some suggestions are split into stages, so need a couple of tab hits. Couldn't the comment be always a separate hit?

**Reply by steinhh** [reply]
Well, it would be useful to me because it explains that styleBinding is not a style binding but an index to/for a style binding🤓. But it should have suggested a change in the variable name rather than an explanation. I wish I could tell Copilot to "please fix up my variable and function names so they are 100% consistent over my entire code base". Hm... I should probably try that!

**Reply by Rippletank** [reply]
Ha, I take your point with the index part, although it's the terminology used by the api so it's appropriate in context.  But there are plenty of other like: 
FontManager fontManger; // For text rendering 
I mean, its not wrong, but it is just noise and make important comments less visible. 
Is there any way to add a global prompt to limit comments to instances of unusual usage?

**Reply by steinhh** [reply]
I agree, it's totally noise. And I always say that a) don't write comments - use extra variables & functions with good names so your program reads like an essay, b) you're allowed to *explain* (we do X *because*...) since that's not something you can put in code, and c) if an AI can comment well, then the comment is probably not needed!

**Comment by steinhh** [comment]
So where's the issue we should vote up? Is there any point in upvoting what we all agree on in here?

**Reply by SPREEKDOS** [reply]
This one 
https://github.com/microsoft/vscode/issues/313003

**Comment by steinhh** [comment]
Oh, right, I use the extension "Disable Copilot Completions in Scopes"!

**Reply by SPREEKDOS** [reply]
sadly seems its not working now :( , but the curse word method at least works.
